lib/Backends/CPU/CPUBackend.cpp

```diff
@@ -451,3 +487,19 @@ llvm::StringRef CPUBackend::getLibjitBitcode() const {
   return llvm::StringRef(reinterpret_cast<const char *>(libjit_bc),
                          libjit_bc_size);
 }
+
+/// Returns true if network supports Type Lowering from T1 to T2.
+/// Allows the backend to control node white/black list for optimization.
+bool CPUBackend::canDoIndexTypeDemotion(
```

Comment:
Well canDoIndexTypeDemotion can rely on isOpSupported. Although I am not sure isOpsupported is representative enough. The way I see it is it answers a question: Does backend X support node Y for specific combination of types. Which worked because we were just operating on one type: float. Now we are operating on two independent types, in two passes. So there is a state in which the answer will be false.

To clarify let's say we have Gather Node. It's input is float and Index of type int64. Now backend X only supports Quantized input and index of type int32. When we query it for quantization we will be asking does it support Quantized and index int64. Because we are only intersted in quantization part. If isOpSupported implemented as it is intended the answer has to be no. Because it only supports index type of 32. 

Similar thing happens when we querry it for index type demotion.

If goal is to eliminate BE from being passed around, maybe a better approach is generatlize canDoIndexTypeDemotion back to canDoTypeDemotion. It then can return a black list of nodes that can't be transformed and it can be used by generic TypeAToTypeBFunctionConverter, and passed to FunctionQuantizer after BE is queried in quantizeFunction.